### ISSUE TITLE
anonymous registration at restaurants, hairdressers etc. using CWA

### ISSUE THREAD

**Issue by hackedal** [corona-warn-app/cwa-wishlist#138]
## Problem and motivation

At the moment there are a lot of negative news in the media about the issue, that personal registration data, which has to be left when visiting restaurants, hairdressers etc. has been abused for example by the police. People are afraid of fraud, tending to leave fake contact data, which makes contact tracing impossible in case of an infection.

## Feature description

It would be great, if CWA could generate (bypassing the contact tracing API) a "manual" random key, maybe 12 oder 16 characters long, that is presented for a limited amount of time (5 minutes should be enough) on the display of the device. This key can now be written on the restaurant owners guest list. After the key disappeared from the display, it is stored for 14 days locally in my device. In case of an infection, the restaurant owner will send his list to the health authorities, who will inform the people on the list. If there are non-CWA-guests, who left their contact data, they have to do it manually but if a CWA-user has left his generated key, it will be added to the central database and fetched by all the CWA-users and the restaurant visitor will be warned, because his device matched it's own generated key.

This feature would have a lot of advantages:

- full privacy for CWA-users: neither the restaurant owner nor the police will ever know, who was in the restaurant
- a warning for CWA-users, even when non-CWA-users were in the same restaurant
- more CWA-users, because only CWA-users will have the advantage of staying anonymous

I hope you like my idea!

---
Internal Tracking ID: EXPOSUREAPP-2034

**Comment by daimpi** [comment]
Related:
- #59 
- #70 
- #77

**Comment by chr-chr** [comment]
Hello,

I've come up the same idea but in a different way.

To my understanding the Corona-App is bound to an user.

Having the App working/registered as a location on a device provided by owners,
they would get an active guest list. 

The Visitor can actively or automatic login and out having full privacy.
Owners can prove their countermeasures.

Regarding the police interests the CWA might help as well:
Actually a backtrace gets done on a positive corona test.
How about giving the police a way to issue a call for witnesses in an anonymous way?

**Comment by Ein-Tim** [comment]
> How about giving the police a way to issue a call for witnesses in an anonymous way?

I don't think thats a good idea (neither do I think the public would appreciate that).
The CWA should **just** be used to inform the user if they had contact with an Covid-19 sick person.

**Comment by DirkReu** [comment]
In #140 I described a little different approach. Both variants avoid having a open tracking of the user's daily life. It would be worth weighing advantages and cost of both ways.
I close #140 to keep the number of requests small.

**Comment by cutec-chris** [comment]
An additional positive effect of this feature is, that more users would use cwa cause this feature makes it a lot easyer to "check in" in restaurants,cinemas and so on.

Additional to the ops idea, the restaurants could have an own app that can scan the code and store it. Maybe those app could additionally "scan" (fotograph) the paper formular of non-cwa users and encrypt it and send all together to the Gesundheitsamt if needed. That would also protect non-cwa users from authorities accessing the paper-registrations. Cause the restaurant yould destroy the paper registrations daily after the app has scanned and encrypted it.

Also the additional app could keep track of deleting the encrypted daily files after 14 days.

**Comment by mf179** [comment]
I like hackedal´s idea. And that it should be enhanced by scanning the code by an App/phone of the restaurant as proposed by cutec-chris.

So the basic functionality would be the same as of today, just that the BT transmission is replaced by code scanning:
- visitor requests CWA to generate a code that is then displayed as QR-code on his phone, visitor´s CWA keeps the code for 14 days
- restaurant´s CWA scans the code, verifies authenticity as a CWA code (to avoid faked QR-codes), confirms reception with some audio signal so that the responsible knows that obligation of registration is fulfilled, and restaurant´s phone/CWA keeps the received code for 14 days

Just the notification process differs as the feature implies a one-way-registration only. An infected visitor can therefore only warn the restaurant´s app (the usual way with sharing his code via server). And the restaurant app (that collected all visitors) needs then to notify the other affected visitors by sharing the related, stored visitors´ codes via the server. The visitors´s apps need to scan the shared codes for their own codes.

I would share Ein-Tim´s concern that everything that is additionally proposed may mainly delay or even hinder the adoption of the basic proposed feature.

It should be noted that while this feature may extend the reach of the CWA the user is still free to deactivate CWA´s BT-contact-tracing, if they don´t like it. Which may further help for adoption. And it works with all phones regardless whether those support the ENF.

**Comment by ferdinand** [comment]
Looks like the UK app has included a „Venue check-in“ feature.

I‘m pretty sure this feature would give the CWA a remarkable boost. Both, venue owners and guests, would love to have a privacy friendly check-in app instead of paper lists and shady third-party software tools.

**Comment by stbrit** [comment]
This would be a game changer. Everyone would get the app if they don't need to fill out paper registrations anymore. And people would be contacted via the app if needed quickly. Such a game changer. Britain managed to do it in their app surely we can as well. This needs to be an absolute priority.

**Comment by gittyhub2018** [comment]
I think this is a good idea. Another idea is, that the Corona App scan a QR Code in the Restaurants or something like else. This QR Code can be dynamic created at a screen in the restaurant or can printed every hour. This code have following information:
- Date and Time in UTC.
- Public Key and cryptographic signing information, that the random key is allowed for CWA using (Spoofing Protection or Protection against illegal hacking from CWA. If you accept too long unsecured strings, you will be hacked if the CWA has a bug).
- random key.
The guests can scan this QR Code and if the guests must be warn, the random key and Date/Time in the QR Code will be transmitted over the CWA Servers. With a recreation of this QR code every hour  you can tell all CWA Users who are effected. The QR Code have not to be changed so fast, because the code is only in the local app and the owner of the restaurant have not to transmit too much keys, after he or she will warn the users.  This could be improve the performance, then you have not be transmit too much keys. Restaurants can change the key every hour e.g with a normal paper print (if the restaurants have not enough technology knowledge or similar things for a Screen.). This can improve the acceptance by the owners.
If the owner need a acknowledge from the CWA, this could be a Bluetooth beacon, who don’t own any personal information from the users. This acknowledgment must not be stored permanently! The MAC Adress or other informations could be have personal information.

This can improve the acceptance of CWA, solve the problem of personal information with guest lists and with a QR Code an owner can easy implement this in his own workflow. An QR Code is easy to create and make the contact tracing safe and easy (KISS-Principles).

**Comment by daimpi** [comment]
I just recently saw someone on [Twitter](https://twitter.com/koos/status/1311037709310459905) suggesting integration of [their registration app](https://github.com/railslove/rcvr-app/) into CWA to provide this functionality 🚀.

**Comment by cutec-chris** [comment]
Yeah a lot of people have this idea, and think its good. But noone of the Devs answers here. Maybe thers not enougth money left ;)

**Comment by spekulatiusmensch** [comment]
There is now a proposal called [**CrowdNotifier**](https://github.com/CrowdNotifier/documents) that seems to fit this issue. It would be interesting how it could be integrated into CWA. They actively ask for feedback and I think it should be discussed here as well.

**Comment by daimpi** [comment]
Linus Neumann also brings up the idea of "Decentralized Presence Tracing" on his blog (German): https://linus-neumann.de/2020/10/die-corona-warn-app-verliert-den-anschluss/

**Comment by rennertimo** [comment]
> 
> 
> Linus Neumann also brings up the idea of "Decentralized Presence Tracing" on his blog (German): https://linus-neumann.de/2020/10/die-corona-warn-app-verliert-den-anschluss/

The proposal that Linux Neumann describes matches #140 .
I like the idea. Easy to implement, make use of existing assets, no additional overhead, same fast reporting in case of a positive test result

**Comment by digital-codes** [comment]
Can we integrate the https://platform.projecttogether.org/initiative/caMqKqhU8rffg1RMqSrMH3ukgPn1 from "wir vs virus" to get the "registration" functionality?

**Comment by MikeMcC399** [comment]
@digital-codes 
> Can we integrate the https://platform.projecttogether.org/initiative/caMqKqhU8rffg1RMqSrMH3ukgPn1 from "wir vs virus" to get the "registration" functionality?

The link which you mention leads to darfichrein.de

My understanding after looking at https://www.darfichrein.de/dir/home including https://www.darfichrein.de/dir/faq is that this solution is not anonymous. In fact after scanning the QR code on https://darfichrein.de/dir/do-it , the user is asked for their name.

So I don't think this could be tightly integrated into CWA, which is based on anonymity and lack of knowledge of physical location, which is fundamental to the GAEN (Google Apple Exposure Notification) Framework / System that CWA is using.

Also darfichrein.de stores data centrally, not on the user's smartphone, which makes it very different to the decentralized way that CWA is stored. It is difficult to image these two aspects coming together as well.

**Comment by cascremers** [comment]
> There is now a proposal called [**CrowdNotifier**](https://github.com/CrowdNotifier/documents) that seems to fit this issue. It would be interesting how it could be integrated into CWA. They actively ask for feedback and I think it should be discussed here as well.

Many ideas above are similar, but for privacy and data protection reasons CISPA would support the decentralized [**CrowdNotifier**](https://github.com/CrowdNotifier/documents) approach mentioned already by @spekulatiusmensch .

**Comment by Thomas-Hendrik** [comment]
Hi, I wrote to the SAP-Team via corona-warn-app.opensource@sap.com yesterday to emphasize the importance of such a functionality. As soon, as I am getting a response, I will write about it at this place.
How can we push it? 
There is one thing I am missing in the features described: The App needs an Application in the backend that helps the Gesundheitsämter to analyse the infection spread patterns and automatically inform people.
How do they do it today? In reports I alway see people (e.g. soldiers) sorting paper and doing phone calls...

**Comment by gittyhub2018** [comment]
> How do they do it today? In reports I alway see people (e.g. soldiers) sorting paper and doing phone calls...
> 
> 

The soldier or health staff doesn't get any information about the app. This is 'cause privacy protection. The high privacy level is the reason, why many people install it and in other countries this doesn't work. CWA should warn the people that they can inform the health office and ask for further information. The health office get their information from doctors and Analyse labs, who test against the virus. Is someone infected they give the address of all possible people to the health office. Only the health office can enforce a quarantine. And in most case, the contact tracing after an infection case is the paperwork. 
But CWA should not tell the information automatically to the health office, because people can have privacy concerns and fight against the CWA using.
For digital contact tracing after infections should be a developed a WebApp, which can automatically connect the same person to a database. CWA should not do this.

I hope now is it more clearer.

**Comment by alanrick** [comment]
> How do they do it today? 

I use the Irish app. Under settings you can add your mobile number so that the tracing service can contact you directly if you want. It’s voluntary. There are over 1.3 million active app users (Ireland has a population of 4.4 million) so the evidence is that this voluntary field helps combat Covid but doesn’t deter users from downloading the app. 
![5176D7F9-3006-476A-8619-57AB5D32E1BB](https://user-images.githubusercontent.com/7034358/98728005-2143ec00-2399-11eb-9c01-183eb1bf10b7.jpeg)
![AB34A9C0-F3E7-412F-9C5D-759510556185](https://user-images.githubusercontent.com/7034358/98728048-31f46200-2399-11eb-9a11-cc2ac6d9f7f5.jpeg)

**Comment by Thomas-Hendrik** [comment]
The Irish App won´t be the mass solution for Germany, let's better push the CrowNotifier solution as an extension of the CWA: https://github.com/CrowdNotifier/documents

**Comment by derhingugger** [comment]
I like the idea to give the CWA a chance to contact me after I visit in a restaurant, cinema or shopping mall in case there was a positive tested person. It make the CWA stonger and more atraktive if this give me the chance to visit these kinds of places. Restaurants or Cinema can open for people they willing to leave these contacts by checking in and out.

**Comment by alanrick** [comment]
And here's the use-case with the Minister for Health (Jens Spahn) in the leading role 😉
https://www.apotheke-adhoc.de/nachrichten/detail/politik/trotz-corona-spahn-informierte-restaurant-nicht/

I suspect that's why he's keen on having the app display the date dangerous contacts took place, but I agree with you that adding this suggestion (#138) would be more effective.

**Comment by Onophrio** [comment]
It is absolutely meaningful, not only to protect personal data from „lying around“ openly on paper in the restaurant, but also to ensure motivation of data security-concerned folks to actually use the corona app. Would be a push to download numbers.
If data is stored on federal servers (voluntarily allowed by app users), this would mean a huge relief of workload in health administration in tracking down infection chains. Even, if not everybody will use that feature. Every piece of paper the „Gesundheitsamt“ does not have to evaluate manually gives them more time to focus on the questionable cases.
In principle, this works, as Bavaria demonstrated by (again) going forward with pragmatic solutions:
https://www.bayern.de/neue-software-vereinfacht-corona-registrierung-bei-gaststaetten-gerlach-lobt-clevere-digitale-loesung/

This is bei. adopted by restaurant in other states, which ashames their state governments as well as federal government.
I would appreciate being able to deliver the full package of data to RKI, given they are stored there encrypted and only RKI and health authority can access it. 
If marked clearly in the app with a toggle „Data plus“ or something, people would dinghaft deliberately and all data protection concerns are meaningless - and I bett people see the usefulness and they WILL use it.

**Comment by alanrick** [comment]
Encouraging statistics have been released for the UK Corona app venue checkin capability.

_...users have checked into a venue over 103 million times. A total of 253 venues were identified as ‘at risk’ as a result of an outbreak since 10 December, triggering ‘warn and inform’ alerts to app users_ 

https://www.gov.uk/government/news/nhs-covid-19-app-alerts-17-million-contacts-to-stop-spread-of-covid-19

**Comment by alanrick** [comment]
> The Irish App won´t be the mass solution for Germany, let's better push the CrowNotifier solution as an extension of the CWA: https://github.com/CrowdNotifier/documents

I heard an interesting discussion with SwissCovid development today. They’ve tried it as a small pilot with CrowdNotifier and are mulling over whether to integrate it.

BTW I wasn’t suggesting using the Irish (or Latvia) App in Germany, but allowing the users to add their mobile number voluntarily by CWA users to alleviate the strain on the tracing health Centers.

**Comment by KaiRoesner** [comment]
There is an app that does a similar thing, the [Luca](https://luca-app.de/) app. It has already officially been adopted in Jena (Thuringia), Salzlandkreis (Saxony-Anhalt), Sylt, Amrum, Föhr (Schleswig-Holstein), Schwerin (Mecklenburg-West Pomerania) and it is being [discussed in Karlsruhe](https://bnn.de/karlsruhe/luca-app-corona-kontakte-nachverfolgung-datenschutz-freie-waehler-landrat) (Baden-Württemberg, BW) and even for state wide adoption in BW. The data protection officer of BW [assesses the app as compliant](https://www.baden-wuerttemberg.datenschutz.de/lfdi-brink-unterstuetzt-nutzung-der-luca-app/) with data protection regulations.

**Comment by dsarkar** [comment]
FYI https://github.com/corona-warn-app/cwa-app-android/pull/2456

**Comment by Sundie1** [comment]
I think the work on this Feature is obsolet becouse the people and the cities use the Luca App which worked fine. 
The Idea to this Feature on the cwa is older then 8 month, that is a realy long time for developmemt. And now the other App is going rollouted. 

But if the cwa will come with this Feature i think the qr Codes should be complatible with the Luca Codes. The cwa should then read Luca qr Codes to Check in. If there is a different qr Code System every place must offer 2 qr Codes and this wont be easy for a enduser to recognize which Code is for Luca wich for cwa. When you scan the wrong Code and it wont work it can be very frustraded for everybody.

**Comment by Ein-Tim** [comment]
@Sundie1 

The problem is that Luca needs personal data (address, name, telephone number, etc.) and CWA not.
So it will get really hard to combine this two systems.

**Comment by alanrick** [comment]
Yep, the Luca app (or recover/darfichrein) is very easy to use with a fantastic UI. But it is merely to satisfy legal regulations. In terms of usefulness I rate the CWA check-in much higher because it is anonymous so there's no micky mouse data to chase up by the overworked track/trace centers. Those alerted by the CWA can get tested without waiting to be contacted.

But @Sundie1 , it is a shame and difficult to understand why the RKI has delayed it's implementation so long. Totally agree with you on that.

**Comment by Thomas-Hendrik** [comment]
In fact, the Luca App is in use in more and more regions due to massive PR. And in fact, the formed letter based registration is far from being compliant (the contact data were sent via E-mail to the health authorities).
Month after month the CWA did not implement the vital functionality only after Luca becam more an more polular ande more and more local politicians decided to use it, the federal government started to react and so did the CWA consortium.
It is a political decision to defend the hardly criticized CWA the support with citizens taxes.

**Comment by Ein-Tim** [comment]
@Thomas-Hendrik 

I totally agree with you. 
Everybody is now setting up the Luca system or has already done so and instead of clearly communicating that CWA will deliver a similar feature which keeps the anonymity (!) of the users, there is no communication about this (outside from GitHub and some articles) at all.

I don't think that it's (only) a political decisions, if you want higher data privacy, CWA is the way to go.

**Comment by KaiRoesner** [comment]
@SebastianWolf-SAP , can this be brought up with RKI to have the priority of this feature adjusted?

**Comment by MikeMcC399** [comment]
@KaiRoesner 
> can this be brought up with RKI to have the priority of this feature adjusted?

What sort of priority change are you asking for? Are you asking for a scope-change compared to the documentation published on https://github.com/corona-warn-app/cwa-documentation/blob/master/event_registration.md? Or do you want a different delivery date for what is already planned?

The press have been reporting that check-in functionality will be released in CWA after Easter https://www.google.com/search?q=corona-warn-app+check-in.

**Comment by dsarkar** [comment]
Dear community,

The current planning is that Event-Registration functionality will be implemented with the next release, version 2.0. If you have specific ideas for feature enhancements regarding check-in functionality, please, open for each specific feature request (enhancement or new feature) a separate issue in this wish-list repository. Thanks!

**Comment by KaiRoesner** [comment]
@dsarkar , @MikeMcC399 , that's very good to hear. It would have been nice if that had been communicated earlier here in this wishlist item.

**Comment by Ein-Tim** [comment]
@KaiRoesner 

@dsarkar posted a link to a PR in the Android repo 27 days ago: https://github.com/corona-warn-app/cwa-wishlist/issues/138#issuecomment-786467315

**Comment by Sundie1** [comment]
Can you please comment if this is correct:

https://blog.wdr.de/digitalistan/corona-warn-app-2-0-kommt-mit-luca-light-inside/

"Es soll möglich sein, mit Warn-App und Luca-App denselben QR-Code zu scannen. Betreiber von Locations müssen also nicht unterschiedliche QR-Codes für die beiden Apps erstellen."

This will be great and exact the usability that i want

This is realy hard to understand

" Und die Luca App für überall dort, wo eine Kontaktnachverfolgung vorgeschrieben ist, etwa in Restaurants, Fitness-Clubs, auf Konzerten. Die Luca App ersetzt den Papierkram, den wir im Sommer vergangenen Jahres stets zu bewältigen hatten."

Why. If this is correct, what usecase has the cwa? The customer will check in in Restaurants without paperwork. And if cwa will NOT be able to handle that, NOBODY will use this Feature.

**Comment by Ein-Tim** [comment]
@Sundie1

*Disclaimer: I'm not a developer or community manager.*

The [documentation](https://github.com/corona-warn-app/cwa-documentation/blob/master/event_registration.md) also mentions a [interoperability](https://github.com/corona-warn-app/cwa-documentation/blob/master/event_registration.md#interoperability-with-other-contact-tracing-apps-draft).  

So it should be possible to scan the same QR-Codes for CWA and Luca, but only if Luca adapts their QR-Code scheme to the one from CWA (this is my understanding). 

I can't say anything to the paperwork thing, sorry.

**Comment by Sundie1** [comment]
(german) hallo, ich finde immernoch kein usecase für diese Funktion so sie wie die umgesetzt ist. Laut Tagesschau "Die anonyme Eventregistrierung der CWA ist hauptsächlich für private Events gedacht", erklärt das Gesundheitsministerium - gemeint sind etwa Familienfeiern oder Konferenzen."

Warum brauch ich bei einem privaten Event eine anonyme Registrierung? Auf meiner Geburtstagsfeier kenne ich doch jeden. 
Bei Konferenzen gibt es so oder so Teilnahmelisten.

@Ein-Tim ich habs mir durchgelesen, aber so wie ich das verstanden habe gibt es hier ein Struktuerelles Problem im Konzept was die Nutzung extrem erschwert. Ich habe das so verstanden, daß im qrcodepayload die start und Endzeit mit codiert wird. Das verhindert doch vollständig, das ich den QR ausdrucken und an die Eingangstür meines Geschäftes oder auf dem Tisch an meinem Biergarten kleben kann.
Normal sollte die Start und Endzeit nicht codiert werden sondern in dem Moment des checkins registriert werden. 
Korrigiert mich wenn ich das falsch verstanden habe.

**Comment by Ein-Tim** [comment]
Hi @Sundie1 

Im Payload des QR-Codes ist die `defaultCheckInLengthInMinutes` dabei, diese ist dafür da das dir dann im der App nach dem Einchecken eine Zeit (die du verstellen kannst!) angezeigt wird. 

Beispiel:
Jemand setzt die `defaultCheckInLengthInMinutes` auf 30 Minuten, du checkst also ein und die CWA schlägt dir vor das du automatisch nach 30 Minuten ausgecheckt wirst. Wenn du mehr/weniger Zeit brauchst, dann stellst du die Zeit ganz einfach über einen Picker in der App ein. 

Noch zu dem Usecase: Dadurch das die Funktion der CWA anonym ist, erfüllt sie (die momentanen) Anforderungen der Verordnungen zur Kontaktnachverfolgung der Länder nicht. Deswegen schreibt die Tagesschau auch das diese Funktion (voraussichtlich) vor allem für den Privaten Gebrauch genutzt werden wird. 
Aber wir werden sehen, vielleicht ändert sich ja was in den Verordnungen...

---

English will follow soon.

**Comment by Sundie1** [comment]
@Ein-Tim Thank you for the explanation 👍

**Comment by thomasaugsten** [comment]
Die Qr-Codes können für beliebige Events verwendet werden wie man auch an der Auswahlliste beim Erstellen sehen kann. Es ersetzt aber nicht die Papierlisten.
Start und End Zeitraum sind nur für zeitlich begrenzte Events relevant.
Dauerhafte Lokations müssen keine Start und End-Zeitraum eintragen aber es ist empfohlen die QR-Codes regelmässig zu ändern aber das ganze System funktioniert trotzdem wenn die QR-Codes nicht geändert werden und festverklebt sind.

**Comment by alanrick** [comment]
I'd also really like to see the use-case for this upcoming feature because even though I supported it I can't figure out how it would help so long as the legal requirement for named-visitor lists remains.

Suppose I visit a concert in Baden Württemberg (where LUCA has purchased).
1. The venue has to print a qrcode which I can scan with the luca app (or I scribble my name in a list).
2. The venue can also (optionally) print a qrcode for the CWA which I can optionally scan. 

I realise that Luca could code their qrcode to support the CWA, but why should they? It's an extra effort and I believe BW purchased the Luca licence without committing Luca to be compatible with the CWA.

Bottom line, what is the motivation for the venue to generate and **print** **both** codes instead of just the Luca qrcode? What is the motivation for the visitor to **scan both** instead of just Luca?

I'd supported this feature because I'd hoped it would give the general public an easily understood motivation to install the CWA in order to simplify their lives (as in the UK app), but the opposite seems the case.

**Comment by alanrick** [comment]
The context... these where the advantages of this feature from the original suggestion. But all 3 are offered by the Luca app (or equivalent such as #darfichrein) and the upcoming CWA doesn't often more.  

> This feature would have a lot of advantages:
> 
> 
> - full privacy for CWA-users: neither the restaurant owner nor the police will ever know, who was in the restaurant
> 
> - a warning for CWA-users, even when non-CWA-users were in the same restaurant
> 
> - more CWA-users, because only CWA-users will have the advantage of staying anonymous
> 

The CWA can only offer an advantage when the BMG retracts the obligation for name-lists to be maintained.

**Comment by ndegendogo** [comment]
@alanrick I don't know details of the solution design of the UK app. How much is it integrated with the health authorities?

> when the BMG retracts the obligation for name-lists

of course this is a **political** decision. But I don't expect this to happen, especially with the rise in all numbers currently.

cwa philosophy is *fully anonymous* and *fully voluntary*. Even when I receive a red warning, it is still my own decision if I take a test, if I self-isolate, or if I just ignore it. And if I am tested positive as a cwa user, I can still decide if I am sharing my keys and warn others or don't (although the latter decision would violate our feeling of fairness).

In an anonymous setting (like cwa) what is possible is to notify / warn others. What is not possible is to **enforce** quarantine or mandatory testing on all (anonymous) participants of an event. But this is what health authorities might need in some cases.

The solution of cwa was designed under time pressure. It is decentral, which contributes imo a lot to the acceptance / adoption of cwa. But it is not integrated with any of the infrastructure of the health authorities.
*(I assume this is different with UK?)*

**Comment by Sundie1** [comment]
In addition to your example with the concert @alanrick
As consumer, on entrance, during i wait, i open the Luca app, my Code is scanned.
Now i can go ahead and close the Luca app... but with the cwa during the Check in i had to switch ne app. (close Luca, open cwa, scan again) this Takes a long time. And nobody understand it. 
On other Locations the same. No consumer will open the first App, scan, switch the App and scan again. 

Why must Luca change their System to Support the cwa qrCode? They are the market leader. The cwa should handle the Luca Codes. 
But becouse there are no usecases for the cwa checkin i think it isnt neccassary to invest development time. If cwa want to beat Luca, the cwa System must be better. 

The cwa is good as it is with the nearby contact tracking.
And the nearby contact tracking with Bluetooth will work on my private Birthday. The usecase Birthday and private Event dont need any qr code checkin.

Now i wait for the cwa checkin Release and wait what the politicans and Newspaper will say.

**Comment by alanrick** [comment]
> @alanrick I don't know details of the solution design of the UK app. How much is it integrated with the health authorities?
...
> *(I assume this is different with UK?)*

I shouldn't have mentioned the UK as this is irrelevant.  But for the record this is is user-anonymous and decentralised, but linked to the health authorities; eg to trigger notifications from a venue (which is not anonymised) 

I'm not suggesting the UK app is technically better, but by offering the convenient qrcode check-in and by not requiring additional apps or paper lists they increased both the usage ratio and cluster detection. 

My struggle is understanding the use-case of this CWA feature, because the advantages described in the initial suggestion no longer apply when the CWA cannot be used alone for venue-checkins, but only in conjunction with a name-checkin app. I don't see how this encourages adoption and the other advantages listed are delivered by the name-checkin app.

**Comment by thomasaugsten** [comment]
Every 3rd party app to replace the paper lists can generate an CWA code in their QR-Code.
The idea ist that the host is generating by default into his QR-Code an CWA-Code. When a user is scanning the qr-code with the fitting digital paper list app like Luca he sees an button to also checkin into the CWA. When he press the button he will get redirected to the CWA and can press checkin to checkin into the location an to receive warning immediately without waiting to get contacted by the Gesundheitsamt
And not all Gesundheitsämter are supported by the Luca at the moment 104 of 400

**Comment by alanrick** [comment]
So the _use-case_ is that:

> I check-in to a concert, and later another member of the audience using the CWA who was outside my bluetooth range is tested positive but I get notified by the app.

And the advantage is none of the three listed in the original suggestion but a 4th:
4) I'm notified earlier than had I waited for the Gesundheitsamt.

Correct?

I'm not convinced this is as significant as if the BMG dropped the obligation for name-lists so that the original 3 advantages were provided by the CWA.

**Comment by alanrick** [comment]
Just to add that the Computer Chaos Club has released a statement about the Luca App.

https://www.ccc.de/de/updates/2021/luca-app-ccc-fordert-bundesnotbremse

I hope this will motivate the authorities to allow anonymous logins via the CWA instead to significantly improve the use-case of the CWA, leading to CWA uptake.

**Comment by MikeMcC399** [comment]
@alanrick 
Wow! Thank you for posting about the Luca app! I am completely shocked 😧 !

Great work on the side of the CWA app though to continue to add value without compromising privacy! I'm looking forward to seeing the release of 2.0 (even though we are in lockdown and I don't see myself checking into any venues the next days!).

**Comment by alanrick** [comment]
> Great work on the side of the CWA app though to continue to add value without compromising privacy!

Exactly. This is revolutionary - not just in theory but in practice. Kudus to all CWA developers. 
I'm just disappointed that the RKI and BMG have been so sluggish at understanding this, indirectly promoting disdain in the general public. A huge lost opportunity.

**Comment by raceface2nd** [comment]
I pointed the anonymized checkin out as well in this [post](https://github.com/corona-warn-app/cwa-wishlist/issues/446#issuecomment-818980219) as the quick test result transfer should work accordingly anonymized.

**Comment by vaubaehn** [comment]
@thomasaugsten @mlenkeit 

> Every 3rd party app to replace the paper lists can generate an CWA code in their QR-Code.
> The idea ist that the host is generating by default into his QR-Code an CWA-Code. When a user is scanning the qr-code with the fitting digital paper list app like Luca he sees an button to also checkin into the CWA. When he press the button he will get redirected to the CWA and can press checkin to checkin into the location an to receive warning immediately without waiting to get contacted by the Gesundheitsamt
> And not all Gesundheitsämter are supported by the Luca at the moment 104 of 400

As we can expect the release 2.0 soon, would you say the specification of the presence tracing QR code data has been finalised?
If so, would be good if you contacted luca, they're somehow waiting ;)
https://gitlab.com/lucaapp/android/-/issues/32

**Comment by vaubaehn** [comment]
> If so, would be good if you contacted luca, they're somehow waiting ;)
The 'draft status' from the Event Registration documentation has been removed, so luca should now be prepared ;)

**Comment by Ein-Tim** [comment]
@vaubaehn 

I [told](https://gitlab.com/lucaapp/android/-/issues/32#note_553987102) the luca people about this (;

**Comment by alanrick** [comment]
Kudus to all developers- Ux, code... The implementation is so beautiful that I've already persuaded several friends who'd deinstalled the CWA to reinstall it. It's a shame it took 9 months to arrive (I guess the delay was in approving the development rather than coding) and I'm still hoping the rules will change to permit anonymous checkins with the CWA to increase the apps usage, and hence effectivity (the principle goal), but even without that - it's a success story. 

A brilliantly implemented feature.

**Comment by Ein-Tim** [comment]
Hey @hackedal, version 2.0, which contains the event-registration, has been released on the 21.04.2021 (see blog post "[Project team releases Corona-Warn-App 2.0 with event registration](https://www.coronawarn.app/en/blog/2021-04-21-corona-warn-app-version-2-0/)"). The releases can be found [here](https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.0.3) for Android and [here](https://github.com/corona-warn-app/cwa-app-ios/releases/tag/v2.0.3) for iOS.
The feature indeed is not exactly the same as what you've been asking for here, but do you think that another solution, besides of the event-registration feature in version 2.0, is needed?
If you see your proposal fulfilled by the event-registration feature, I'd like to ask you if you could close this issue. 

Thank you!

**Comment by Sundie1** [comment]
German:
Ich finde die Umsetzung und das look&feel der V2.0 bezüglich der QR Code Erfassung gut gelungen.
Stringend wäre es ja wenn der Nutzer sich jetzt aussuchen kann. Entweder CWA oder Papierliste.
Das können aber nicht die cwa Entwickler machen. ABER die CWA Entwickler haben in einer Bundeseigenen App eine Funktion gebaut um eine andere Vorgabe die vom Bund kommt zu digitalisieren (digitalisieren ist ja das Wort schlecht hin bei den Politikern) das bedeutet diese Funktion sollte doch jetzt auch den eigenen Gesetzlichen Vorgaben zur Kontaktnachverfolgung entsprechen.
I know das ist hier in github nicht der richtige Ort für diese Diskussion. Wollte es dennoch mal aufschreiben, denn der Ursprungsgedanke war ja wenn ich das richtige deute a) anonymisierung der Registrierung b) Abschaffung der Zettel. (a ist erledigt) ;-)

**Comment by jucktnich** [comment]
Just a question (don't know if this is the right place). What happens if a person, who don't uses CWA got tested positive, can the Event owner give a warning to the CWA users? And if this is possible in the app (I didn't found something like that) how could you make this possible for the website?

**Comment by Ein-Tim** [comment]
@jucktnich 

This is not possible currently. Feel free to open an enhancement request.

**Comment by heinezen** [comment]
Hello everyone,

This feature has been implemented in CWA 2.0 as part of the event registration feature.

This issue will be closed. If you have new ideas for the current implementation of the feature, please consider opening a new issue.

----
Corona-Warn-App Open Source Team
